### PR TITLE
feat(notification-center): build the web component to umd module

### PR DESCRIPTION
**Pull request by LetItRock**
### What change does this PR introduce? 

The idea here is that we will build the Notification Center Web Component to the UMD module and will host it on the CDN.
It will be possible to include it in any web application through the `script` tag, without the hassle of the iframe.

Example:
```
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <script src="https://cdn.novu.com/notification-center/index.js"></script>
    <title>Document</title>
  </head>
  <body >
      <notification-center-component
        style="display: inline-flex"
        application-identifier="YOUR_APP_ID"
        subscriber-id="YOUR_SUBSCRIBER_ID"
      ></notification-center-component>
      <script>
        // here you can attach any callbacks, interact with the web component API
        let nc = document.getElementsByTagName('notification-center-component')[0];
        nc.onLoad = () => console.log('hello world!');
      </script>
  </body>
</html>

```

### Other information (Screenshots)

Screenshot 2022-12-07 at 00 09 00

### CODE REVIEW

**Comment by LetItRock** [comment]
@scopsy WDYT about this?

File: packages/notification-center/rollup.config.js [hunk 1]
```diff
@@ -32,6 +34,29 @@ export default [
...
+  {
+    input: 'src/web-component.ts',
+    external: ['react@17.0.2', 'react-dom@17.0.2'],
+    output: [
+      {
+        file: packageJson.browser,
+        format: 'umd',
+        name: 'NotificationCenterWebComponent',
+        sourcemap: true,
+      },
```
Comment (LetItRock) [inline packages/notification-center/rollup.config.js:46]: the UMD module rollup configuration, the file will be built to the path from `package.json` - `browser` param

File: packages/notification-center/rollup.config.js [hunk 2]
```diff
@@ -32,6 +34,29 @@ export default [
...
+    ],
+    plugins: [
+      replace({
+        'process.env.NODE_ENV': JSON.stringify('production'),
+      }),
```
Comment (LetItRock) [inline packages/notification-center/rollup.config.js:51]: the react and other libs are using `process.env.NODE_ENV` variable, by default Rollup doesn't include it

File: packages/notification-center/rollup.config.js [hunk 3]
```diff
@@ -32,6 +34,29 @@ export default [
...
+      resolve({ preferBuiltins: false, browser: true }),
+      typescript({ tsconfig: './tsconfig.json' }),
+      commonjs(),
+      terser(),
+      gzipPlugin(),
```
Comment (LetItRock) [inline packages/notification-center/rollup.config.js:56]: minify, uglify, gzip

File: packages/notification-center/rollup.config.js [hunk 4]
```diff
@@ -32,6 +34,29 @@ export default [
...
+        sourcemap: true,
+      },
+    ],
+    plugins: [
+      replace({
+        'process.env.NODE_ENV': JSON.stringify('production'),
+      }),
+      resolve({ preferBuiltins: false, browser: true }),
+      typescript({ tsconfig: './tsconfig.json' }),
+      commonjs(),
```
Comment (LetItRock) [inline packages/notification-center/rollup.config.js:54]: convert commonjs modules to es6 so they can be used by rollup

File: packages/notification-center/src/web-component.ts [hunk 5]
```diff
@@ -0,0 +1,3 @@
+import { NotificationCenterWebComponent } from './components/notification-center-component';
+
+customElements.define('notification-center-component', NotificationCenterWebComponent);
```
Comment (LetItRock) [inline packages/notification-center/src/web-component.ts:3]: this is the entry point for the UMD module

**Comment by p-fernandez** [comment]
For reference to anyone like me that wasn't aware what UMD was: https://github.com/umdjs/umd#readme

File: packages/notification-center/src/hooks/use-auth.hook.ts [hunk 6]
```diff
@@ -1,8 +1,15 @@
...
+import { IAuthContext, ISubscriberJwt } from '../shared/interfaces';
 
-export function useAuth() {
+export function useAuth(): {
+  token: string;
+  applyToken: (token: string) => void;
+  user: ISubscriberJwt;
+  setUser: (userr: ISubscriberJwt) => void;
+  isLoggedIn: boolean;
+} {
```
Comment (p-fernandez) [inline packages/notification-center/src/hooks/use-auth.hook.ts:12]: Helpful! 👏🏻

**Review by p-fernandez — APPROVED** [review]
🌟

**Pull request merged**